x/market/types/id.go

```diff
@@ -101,7 +101,7 @@ func (id BidID) OrderID() OrderID {
 
 // String method for consistent output.
 func (id BidID) String() string {
-	return sdkutil.FmtBlockID(&id.Owner, &id.DSeq, &id.GSeq, &id.OSeq, nil)
+	return sdkutil.FmtBlockID(&id.Owner, &id.DSeq, &id.GSeq, &id.OSeq, &id.Provider)
```

Comment:
did this not affect any tests?  seems like `parse(id.String()).Provider == id.Provider` should have failed.

Reply:
I'm a little confused about that too, tests against strings weren't 100% though.

Reply:
The tests were supposed to have a "`decode(encode(x)) == x`" test for all types.  They clearly don't or are broken in some other way.  Fix the test for this type and look through the tests for the other types for similar problems.

Reply:
created an issue to track: https://github.com/ovrclk/akash/issues/752

Reply:
It's because you're not testing `id.String()` [here](https://github.com/ovrclk/akash/blob/26e321d3d2e282a437b382d4b2fc725b8a3e5ef2/sdkutil/object_id_test.go#L231)

Reply:
great catch, updated test and referenced new issue to take a refreshed look at it.